aws/data_source_aws_ebs_volume_ids.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"errors"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func dataSourceAwsEbsVolumeIDs() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEbsVolumeIDsRead,
+		Schema: map[string]*schema.Schema{
+			"filter": ec2CustomFiltersSchema(),
+
+			"tags": tagsSchemaComputed(),
+
+			"ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEbsVolumeIDsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	req := &ec2.DescribeVolumesInput{}
+
+	if tags, tagsOk := d.GetOk("tags"); tagsOk {
+		req.Filters = append(req.Filters, buildEC2TagFilterList(
+			keyvaluetags.New(tags.(map[string]interface{})).Ec2Tags(),
+		)...)
+	}
+
+	if filters, filtersOk := d.GetOk("filter"); filtersOk {
+		req.Filters = append(req.Filters, buildEC2CustomFilterList(
+			filters.(*schema.Set),
+		)...)
+	}
+
+	if len(req.Filters) == 0 {
+		req.Filters = nil
+	}
+
+	log.Printf("[DEBUG] DescribeVolumes %s\n", req)
+	resp, err := conn.DescribeVolumes(req)
+	if err != nil {
+		return err
```

Comment:
It would be great to return error context here for operators and code maintainers:

```suggestion
		return fmt.Errorf("error describing EC2 Volumes: %w", err)
```